### PR TITLE
Pcg 9 feature epic plant list

### PR DESCRIPTION
**Pull request by VitiukVV**
add show list of plants in Garden page

add modal to adding plants to your garden

add contextAPI of list of plants

### CODE REVIEW

File: src/components/App.tsx [hunk 1]
```diff
@@ -1,21 +1,62 @@
-import { lazy } from 'react';
+import { createContext, lazy, useEffect, useState } from 'react';
 import { Route, Routes } from 'react-router-dom';
 import AppBar from './appBar/AppBar';
 import Home from './pages/home/Home';
 
 const Details = lazy(() => import('./pages/details/Details'));
 const Garden = lazy(() => import('./pages/garden/Garden'));
 
+export type DataItem = {
```
Comment (makandriival) [inline src/components/App.tsx]: types and interfaces better keep in a separate folder in src/types or src/interfaces
if you are defining the type for an object its better to use interface

File: src/components/App.tsx [hunk 2]
```diff
@@ -1,21 +1,62 @@
...
+  removePlant: () => {},
+});
+
 const App = () => {
+  const [data, setData] = useState<DataItem[]>(() => {
+    const savedData = localStorage.getItem('plantsData');
+    return savedData ? JSON.parse(savedData) : [];
+  });
+
+  useEffect(() => {
```
Comment (makandriival) [inline src/components/App.tsx:33]: its nice to call useEffect right before the return, improves readability

File: src/components/form/AddPlant.tsx [hunk 3]
```diff
@@ -0,0 +1,128 @@
+import { ChangeEvent, FormEvent, useContext, useState } from 'react';
+import { plantBotanicNames } from '../../data/plantBotanicNames';
+import { rooms } from '../../data/rooms';
+import sprite from '../../icons/symbol-defs.svg';
+import { PlantsList } from '../App';
+
+type AddPlantProps = {
```
Comment (makandriival) [inline src/components/form/AddPlant.tsx]: i usually just call this type Props
its not like you gonna use it elswhere, so there is no need for unique naming
also better use interface

File: src/components/form/AddPlant.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+  );
+  const [selectedRoom, setSelectedRoom] = useState<string>(rooms[0]);
+
+  const plantsListData = useContext(PlantsList);
+
+  const handleSelect = (
+    e: ChangeEvent<HTMLSelectElement | HTMLInputElement>
+  ): void => {
+    const { name } = e.target;
+    switch (name) {
```
Comment (makandriival) [inline src/components/form/AddPlant.tsx:24]: bad practice to leave the default case empty, you can use it for one of the cases you have or at least throw an error or make console.warn so if something go wrong you gonna know what happened
IMHO :)

**Review by makandriival — APPROVED** [review]
i didn't go deep ( did not run it locally as well )
looks good to me, however i left some comments
will approve still, its up to you if you wanna fix them or not ;) 

also might be a good idea to create PrimaryButton component that all of your teammates will reuse around the app, i assume its pretty much the same styling everywhere

**Review by irinamishchenko — APPROVED** [review]

File: src/data/plantBotanicNames.ts [hunk 5]
```diff
@@ -0,0 +1,32 @@
+export const plantBotanicNames: string[] = [
```
Comment (YaroslavRebrina) [inline src/data/plantBotanicNames.ts:1]: why do we have to describe all existing plant names, but don`t do this property as smthing like 
`plantName: string  `?

File: src/data/plantBotanicNames.ts
Comment (YaroslavRebrina) [inline src/data/plantBotanicNames.ts:1]: why to not fetch all data for the first time, save it, and use? And if so we will get rid of this huge, and unnecessary array. (All this data we have on BE)

**Pull request merged**